virtual_rainforest/core/grid.py

```diff
@@ -446,3 +429,129 @@ def populate_distances(
+        """Map a set of coordinates onto grid cells.
+
+        This function loops over points defined by pairs of x and y coordinates and maps
+        the coordinates onto the cell_ids of the grid. The method also checks to see
+        that each point intersects one and only one of the cell polygons defined in the
+        grid. Points that intersect no cells fall outside the grid polygons and points
+        that intersect more than one cell fall ambiguously on cell borders.
+
+        Args:
+            x_coords: A numpy array of x coordinates of points that should occur within
+                grid cells.
+            y_coords: A similar and equal-length array providing y coordinates.
+
+        Returns:
+            A list of lists showing the cell ids that map onto each point. The list for
+            a given point can be empty - when the point falls in no cell - or of length
+            > 1 when a point falls on adjoining cell boundaries.
+        """
+
+        if (x_coords.ndim != 1) or (y_coords.ndim != 1):
+            raise ValueError("The x/y coordinate arrays are not 1 dimensional")
+
+        if x_coords.shape != y_coords.shape:
+            raise ValueError("The x/y coordinates are of unequal length")
+
+        # Get shapely points for the coordinates
+        xyp = [Point(x, y) for x, y in zip(x_coords, y_coords)]
+
+        # Map the Cell ID of each point - this is doing all pairs of points and cells,
+        # which is a computational choke point. Might be able to use a spatial search
+        # option, if this gets problematic, possibly including an STRTree in the grid
+        #    object https://shapely.readthedocs.io/en/latest/strtree.html
+
+        return [
+            [id for id, ply in zip(self.cell_id, self.polygons) if ply.intersects(pt)]
+            for pt in xyp
+        ]
+
+    def map_xy_to_cell_indexing(
+        self,
+        x_coords: np.ndarray,
+        y_coords: np.ndarray,
+        x_idx: Optional[np.ndarray],
+        y_idx: Optional[np.ndarray],
+        strict: bool = True,
+    ) -> tuple[np.ndarray, np.ndarray]:
+        """Returns indexing to map xy coordinates a single cell_id axis.
+
+        This function maps the provided one-dimensional set of x and y points onto the
+        grid (using `~virtual_rainforest.core.grid.Grid.map_xy_to_cell_id`) and then
+        checks that the mapped points uniquely identify all of the grid cells. If
+        `strict` is set to False, then extra points outside the grid are permitted.
+
+        The function then returns a pair of arrays that give indices on the original
+        x and y data to extract data along a single cell id axis. If specific x and y
+        indexing data is not provided, they are assumed to be simple sequences along
+        `x_coords` and `y_coords`.
+
+        Args:
+            cell_map: An existing mapping from points onto cell ids.
+            strict: Should points outside the grid be allowed
```

Comment:
Does this need to be an option? When would you want to call this with `strict==False`?

Reply:
To allow people to use a subset of a larger raster file. I think I know what you're going to say to that though 😄

Reply:
Actually, it is currently only ever used with `strict=False` to allow for data with a larger extent. But that falls under the same heading of putting the responsibility on the user to get the data right, not just approximately right. @vgro - same deal, right?

Reply:
I guess so yes. We would expect users to provide data with the correct extent because we don't want to do the quality check internally

Reply:
But I already did 😭 

However, we should be consistently strict. As we've discussed, we will very likely provide tools to help set files up _outside_ of the `virtual_rainforest` codebase, so we will be supporting people to get the files right.